lib/Runtime/Library/TypedArray.cpp

```diff
@@ -2252,40 +2213,59 @@ namespace Js
         Assert(!(callInfo.Flags & CallFlags_New));
         CHAKRATEL_LANGSTATS_INC_BUILTINCOUNT(TypedArray_Prototype_sort);
 
+        // Throw if not a valid typed array
+        // This step is per spec and allows us to optimise significantly below
         TypedArrayBase* typedArrayBase = ValidateTypedArray(args, scriptContext, _u("[TypedArray].prototype.sort"));
```

Comment:
This is in the wrong place per spec - if we do `Uint8Array.prototype.sort.call(5,5);` per spec the error for the invalid comparison function should be thrown before checking if it's an invalid typed array, currently we'll throw the wrong error.